README.md

# react-native-carrot-design
This is the React Native custom UI components library, It's components for android and iOS.

## Getting started

`$ npm install react-native-carrot-design --save`

## Usage
`javascript`
**Example import**
We provide multiple UI components that you can import on demand, such as:

```
import { ButtonView, CornerTextView, LoadingView, AlertView } from 'react-native-carrot-design';
```

**ButtonView:**
All you need is to import { ButtonView } from the react-native-carrot-design module and then use the <ButtonView/> tag.
```
import React, { Component } from 'react';
import { ButtonView } from 'react-native-carrot-design';
...
<ButtonView activeOpacity={0.6} btnTitleStyle={{color:'blue',fontSize:19}} 
     btnViewStyle={{backgroundColor:'red'}} title='自定义按钮' onPressCallback={()=>{
          alert('pressBtn!');
     }}>
</ButtonView>
...
```

**CornerTextView:**
```
<CornerTextView cornerStyle={{backgroundColor:'orange',width:89}} title="title" textStyle={{color:'red'}}></CornerTextView>
```

**LoadingView**

*note:*
**The LoadingView tag must be the last child of the outermost layout.**

```
...
render() {
    return (
      <View style={styles.container}>
        ...
        ...
        <LoadingView ref={r=>{this.Loading = r}} hide = {true} />
      </View>
    );
 }
  ...
  
```
```
  
//TODO:Call show() when you want to show loading
this.Loading.show();
  
//TODO:Call close() when you want to close loading
this.Loading.close();
  ...
  
  
```

**AlertView**

```
...
render() {
    return (
      <View style={styles.container}>
        ...
        <AlertView
            title="修改姓名"
            contentType={AlertView.DialogType.INPUT}
            content="请输入姓名"
            mainColor="red"
            alertCallback={(text) => {
                newUserName = text;
            }}
            okText="确定啦"
            cancelText="取消"
            onOk={() => {
                this.alertView.setAlertVisible(false);
            }}
            onCancel={() => {
                this.alertView.setAlertVisible(false);
                newUserName = '';
            }}
            ref={(alertView) => {
                this.alertView = alertView;
            }}
        />
      </View>
    );
 }
...
```
```
//TODO:Call setAlertVisible() when you want to set the display of the alert
this.alertView.setAlertVisible(visible:boolean);
```
**BannerView**

```
render() {
        return (
            <View style={styles.container}>
                {/* <BannerView */}
                {/* duration={2000} */}
                {/* pageIndexType={1} */}
                {/* pageIndexNormalImg={normalIcon} */}
                {/* pageIndexSelectImg={selectIcon} */}
                {/* bannerDefaultImg={defaultIcon} */}
                {/* bannerCache="force-cache" */}
                {/* bannerData={ImageData} */}
                {/* bannerKey="img" */}
                {/* onPressCallback={item => this.touchBanner(item)} */}
                {/* /> */}
                <BannerView
                    pageIndexType={0}
                    pageIndexNormalColor="#E8E8E8"
                    pageIndexSelectColor="tomato"
                    bannerDefaultImg={defaultIcon}
                    bannerStyle={{ width, height: 100 }}
                    bannerImageStyle={{ width, height: 100 }}
                    pageIndexViewMoreStyle={{ width, height: 25 }}
                    bannerData={ImageData}
                    bannerKey="img"
                    bannerCanScroll={true}
                    onPressCallback={item => this.touchBanner(item)}
                />
            </View>
            
        touchBanner = (item) => {
             console.log(`数据-打印：${item.index}`);
        }
```

**CollectionView**
```
const headerSource = [{ title: '第一组头部标题' }, { title: '第二组头部标题' }];
const dataSource = [
 [
    {key: '第一组--第一个item' }, {key: '第一组--第二个item', {key: '第一组--第三个item'}
], 
[
   // It is empty section
], [
   {key: '第三组--第一个item' }, {key: '第三组--第二个item'}]
]
...
render() {
    return (
      <View style={styles.container}>
        ...
        <CollectionView ref={r => { this.CollectionView = r }}
          headerSource={headerSource}
          dataSource={dataSource}
          renderItem={this.renderItem} 
          sectionStyle={styles.sectionViewStyle} 
          itemStyle={styles.itemViewStyle}
          renderHeaderComponent={this.rendHeaderComponent}
          renderFooterComponent={this.rendFooterComponent}
          renderEmptyComponent={this.rendEmptyComponent}
          renderSectionHeader={this.renderSectionHeader}
          renderSectionFooter={this.renderSectionFooter} 
          renderSectionSeparator={this.renderSectionSeparator} 
          renderSectionEmptyComponent={this.renderSectionEmptyComponent}
          renderItemSeparator={this.renderItemSeparator}
          stickySectionHeadersEnabled={true}
          initialNumToRender={20}
          onEndReached={this.onEndReached}
          onEndReachedThreshold={0.5}
          inverted={false}
        >
        </CollectionView>
        ...
      </View>
    );
 }
  ...
  ```
```

/**
   * @description: Renderer for every item in every section.
   * @param {item}: The dataSource of the item.
   * @param {sectionIndex}: The index of the section of the current item.
   * @param {itemIndex}: The index of the current item.
   * @return: React.ReactElement | null
   */
  renderItem = (item, sectionIndex, itemIndex) => (
    <TouchableOpacity
      activeOpacity={0.8}
      onPress={() => {
        alert(`I'm ${item.key}`);
      }}
    >
      <Text style={styles.rowText}>{item.key}</Text>
    </TouchableOpacity>
  )

  /**
   * @description: Rendered at the top of each section. Sticky headers are not yet supported.
   * @param {headerItem}: The dataSource of the header.
   * @param {sectionIndex}: The index of the section
   * @return: React.ReactElement | null
   */
  renderSectionHeader = (headerItem, sectionIndex) => {
    return (
      <Text style={styles.sectionHeader}>第{sectionIndex}组:sectionHeaderComponent</Text>
    )
  }

  /**
   * @description: Rendered at the bottom of each section.
   * @return: React.ReactElement | null
   */
  renderSectionFooter = (sectionIndex) => {
    return (
      <Text style={styles.sectionFooter}>第{sectionIndex}组:sectionFooterComponent</Text>
    )
  }

  /**
   * @description: Rendered when the section is empty.
   * @return: React.ReactElement | null
   */
  renderSectionEmptyComponent = () => {
    return (<Text style={styles.listEmpty}>sectionEmptyComponent</Text>)
  }
  ```
**PictureBrowserView**
```
render() {
        return (
            <View style={styles.container}>
               <PictureBrowserView
                                 modalVisible={showPic}
                                 browserData={ImageData}
                                 browserKey="img"
                                 browserNameKey="title1"
                                 isHideSavePhotoView={showSaveAlter}
                                 currentTapIndex={currentIndex}
                                 onSaveCallback={item => this.savePic(item)}
                                 onPressCallback={item => this.touchPic(item)}
                             /> 
            </View>
```

## More detailed subcomponent documentation
[ButtonView doc](https://github.com/rocket-developer/react-native-carrot-design/blob/master/docs/ButtonView.md)

[CornerTextView doc](https://github.com/rocket-developer/react-native-carrot-design/blob/master/docs/CornerTextView.md)

[LoadingView doc](https://github.com/rocket-developer/react-native-carrot-design/blob/master/docs/LoadingView.md)

[AlertView doc](https://github.com/rocket-developer/react-native-carrot-design/blob/master/docs/AlertView.md)

[BannerView doc](https://github.com/aerozhonghuan/react-native-carrot-design/blob/develop/docs/BannerView.md)

[CollectionView doc](https://github.com/rocket-developer/react-native-carrot-design/blob/master/docs/CollectionView.md)

[PictureBrowserView doc](https://github.com/aerozhonghuan/react-native-carrot-design/blob/master/docs/PictureBrowserView.md)
 


CHANGELOG.md
<!--
 * @Description: Changelog
 * @Author: wanglh
 * @Date: 2019-08-19 17:11:09
 * @LastEditors: wanglh
 * @LastEditTime: 2020-06-03 17:17:48
 -->
### Changelog
####v1.1.18 date: 2021-11-01
fix PictureBrowserView style .
####v1.1.17 date: 2021-11-01
fix PictureBrowserView style .
####v1.1.16 date: 2020-06-03
Add TabSelectView, which can customize styles.
####v1.1.15
date: 2019-12-25
*  fix the bug of Banner .
####v1.1.14
date: 2019-12-3
*  fix the bug of Banner and PictureBrowserView.
####v1.1.13
date: 2019-12-3
*  fix the bug of Banner and PictureBrowserView.
####v1.1.12
date: 2019-12-2
*  fix the bug of Banner and PictureBrowserView.
####v1.1.11
date: 2019-12-2
*  fix the bug of Banner and PictureBrowserView.

####v1.1.10
date: 2019-8-30
*  fix the bug of Banner about Android in some type phone .

####v1.1.9
date: 2019-8-30
*  fix the bug of CollectionView data update not refreshing.

####v1.1.8
date: 2019-8-19
*  fix the bug of CollectionView data update not refreshing.

####v1.1.7
date: 2019-7-22
*  fix PictureBrowserView defualt iamge 's URL.

####v1.1.6
date: 2019-4-25
*  Add a new feature to the TextInputView to set the icon on the left.

####v1.1.5
date: 2019-4-23
*  Fix the problem that the CollectionView data source is not cleared.

####v1.1.4
date: 2019-4-23
*  Optimize the section layout of the CollectionView.

####v1.1.3
date: 2019-4-22
*  Optimize CollectionView.

####v1.1.2
date: 2019-4-22
*  Add PictureBrowserView, which can customize styles.

####v1.1.1
date: 2019-4-22
*  Add PictureBrowserView, which can customize styles.

####v1.1.0 
date: 2019-4-18
*  Add CollectionView, which can customize styles.

####v1.0.10 
date: 2019-4-10
*  Add BannerView export, which can customize styles.

####v1.0.9 
date: 2019-4-10
*  Add BannerView export, which can customize styles.

####v1.0.8 
date: 2019-4-10
*  Add BannerView, which can customize styles.

####v1.0.7 
*  fix few bugs AlertView.

####v1.0.6 
*  Add AlertView, which can customize styles.

####v1.0.5 
*  Add TextInputView, which can customize styles.

####v1.0.4
* Fix bug for LoadingView.

####v1.0.3
*  Add LoadingView, which provides display and hide methods.

####v1.0.2
* Edit index.d.ts for CornerTextView.

####v1.0.1
* Add CornerTextView, which can customize styles.

####v1.0.0
*  Add ButtonView, which can customize styles.
  


example/demoList.js
/*
 * @Description: config of all screens
 * @Author: wanglh
 * @LastEditors: wanglh
 * @Date: 2019-04-18 17:28:56
 * @LastEditTime: 2020-06-03 15:25:45
 */

// 一级页面  demo列表
export const MAINSCREENS = [
    {
        name: 'ButtonView',
        title: 'ButtonView',
        description: '按钮',
        module: require('./src/screens/ButtonView'),
    },
    {
        name: 'TextInputView',
        title: 'TextInputView',
        description: '输入框',
        module: require('./src/screens/TextInputView'),
    },
    {
        name: 'CollectionView',
        title: 'CollectionView',
        description: '九宫格',
        module: require('./src/screens/collectionView/CollectionView'),
    },
    {
        name: 'BannerView',
        title: 'BannerView',
        description: 'banner展示',
        module: require('./src/screens/BannerView'),
    },
    // {
    //     name: 'PictureBrowserView',
    //     title: 'PictureBrowserView',
    //     description: 'PictureBrowserView展示',
    //     module: require('./src/screens/PictureBrowserView'),
    // },
    {
        name: 'CornerTextView',
        title: 'CornerTextView',
        description: '圆角文本',
        module: require('./src/screens/CornerTextView'),
    },
    {
        name: 'LoadingView',
        title: 'LoadingView',
        description: 'loading加载',
        module: require('./src/screens/LoadingView'),
    },
    {
        name: 'AlertView',
        title: 'AlertView',
        description: '自定义AlertView',
        module: require('./src/screens/AlertView'),
    },
    {
        name: 'TabSelectView',
        title: 'TabSelectView',
        description: '自定义TabSelectView',
        module: require('./src/screens/TabSelectView'),
    },
];

// 二级页面
// CollectionView
export const COLLECTIONS = [
    {
        name: 'CollectionViewF',
        title: '九宫格',
        description: '经典九宫格',
        module: require('./src/screens/collectionView/CollectionViewF'),
    },
    {
        name: 'CollectionViewS',
        title: '九宫格',
        description: '更多属性',
        module: require('./src/screens/collectionView/CollectionViewS'),
    },
    {
        name: 'CollectionViewT',
        title: '普通列表',
        description: '实现SectionList分组列表效果',
        module: require('./src/screens/collectionView/CollectionViewT'),
    },
];
